modules/RTC/TraceablePeerConnection.js

```diff
@@ -1123,117 +1130,80 @@ TraceablePeerConnection.prototype._removeRemoteTrack = function(toBeRemoved) {
 };
 
 /**
- * Returns a map with keys msid/mediaType and <tt>TrackSSRCInfo</tt> values.
- * @param {RTCSessionDescription} desc the local description.
- * @return {Map<string,TrackSSRCInfo>}
+ * Processes the local SDP and creates an SSRC map for every local track.
+ *
+ * @param {string} localSDP - SDP from the local description.
+ * @returns {void}
  */
-TraceablePeerConnection.prototype._extractSSRCMap = function(desc) {
-    /**
-     * Track SSRC infos mapped by stream ID (msid) or mediaType (unified-plan)
-     * @type {Map<string,TrackSSRCInfo>}
-     */
+TraceablePeerConnection.prototype._processAndExtractSourceInfo = function(localSDP) {
     const ssrcMap = new Map();
 
-    /**
-     * Groups mapped by primary SSRC number
-     * @type {Map<number,Array<SSRCGroupInfo>>}
-     */
-    const groupsMap = new Map();
-
-    if (typeof desc !== 'object' || desc === null
-        || typeof desc.sdp !== 'string') {
-        logger.warn('An empty description was passed as an argument');
-
+    if (!localSDP || typeof localSDP !== 'string') {
         return ssrcMap;
     }
-
-    const session = transform.parse(desc.sdp);
-
-    if (!Array.isArray(session.media)) {
-        return ssrcMap;
-    }
-
-    let media = session.media;
-
-    media = media.filter(mline => mline.direction === MediaDirection.SENDONLY
+    const session = transform.parse(localSDP);
+    const media = session.media.filter(mline => mline.direction === MediaDirection.SENDONLY
         || mline.direction === MediaDirection.SENDRECV);
 
-    let index = 0;
-
-    for (const mLine of media) {
-        if (!Array.isArray(mLine.ssrcs)) {
-            continue; // eslint-disable-line no-continue
-        }
+    if (!Array.isArray(media)) {
+        return;
+    }
 
-        if (Array.isArray(mLine.ssrcGroups)) {
-            for (const group of mLine.ssrcGroups) {
-                if (typeof group.semantics !== 'undefined' && typeof group.ssrcs !== 'undefined') {
-                    // Parse SSRCs and store as numbers
-                    const groupSSRCs = group.ssrcs.split(' ').map(ssrcStr => parseInt(ssrcStr, 10));
-                    const primarySSRC = groupSSRCs[0];
+    for (const localTrack of this.localTracks.values()) {
```

Comment:
`localTracks` is already an array, you can iterate it without the `.values()`

Reply:
Isn't it a Map?

Reply:
Sorry, you're right. I was reading RTC.js where it's an Array 😅